### PR TITLE
Remove duplicate hosts from the etcd machine cache

### LINKED ISSUE #2126: etcd machines_cache can have duplicate entries
**Describe the bug**

While researching retries in case of DCS failures, I noticed that `_calculate_timeouts()` gets 7 as `etcd_nodes` when I use etcd, but only 3 if I use etcd3. In both cases, etcd is running locally on a 3-node Patroni cluster, and all three nodes are mentioned in the `etcd:` (or `etcd3`) section.

Not sure where the difference is from, but `machines_cache()` for `etcd` gets three times three nodes, and removes only the duplicate `base_uri` (the etcd leader it seems) hosts, not the others, so 7 remain.

This implies that `retry_timeout` with  a 3-node etcd cluster is divided by 7 for `etcd` and only 3 for `etcd3`, which looks wrong to me? On the other hand, it is not a big deal.

**Expected behavior**

Same timeout for etcd and etcd3.

**Environment**
- Patroni version: 2.1.1
- PostgreSQL version: 14.1
- DCS (and its version): etcd-3.3.25

**Patroni configuration file**
```
etcd:
  hosts:
    - 10.0.3.195:2379
    - 10.0.3.237:2379
    - 10.0.3.178:2379
```

### PR DESCRIPTION
**Pull request by mbanck**
Fixes #2126

### CODE REVIEW

File: patroni/dcs/etcd.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def _get_members(self, base_uri, **kwargs):
     @property
     def machines_cache(self):
         base_uri, cache = self._base_uri, self._machines_cache
-        return ([base_uri] if base_uri in cache else []) + [machine for machine in cache if machine != base_uri]
+        return ([base_uri] if base_uri in cache else []) + list(set([machine for machine in cache if machine != base_uri]))
```
Comment (CyberDem0n) [inline patroni/dcs/etcd.py:168]: IMO, https://github.com/zalando/patroni/blob/256a359a1ecd60ae7b81e1ac6524e1c85d2e78bd/patroni/dcs/etcd.py#L187 is a better place for it.

Comment (mbanck) [reply patroni/dcs/etcd.py:168]: Fair enough, I have changed it

**Comment by FxKu** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**